Lab4/src/ca/mcgill/ecse211/lab4/Lab4.java

```diff
@@ -0,0 +1,141 @@
+package ca.mcgill.ecse211.lab4;
+
+import lejos.hardware.Button;
+import lejos.hardware.ev3.LocalEV3;
+import lejos.hardware.lcd.TextLCD;
+import lejos.hardware.motor.EV3LargeRegulatedMotor;
+import lejos.hardware.port.Port;
+import lejos.hardware.sensor.EV3UltrasonicSensor;
+import lejos.hardware.sensor.SensorModes;
+import lejos.robotics.SampleProvider;
+
+/**
+ * This is the main class for our program. It starts the program
+ * and it allows the user to select different options on how they want the run 
+ * to work (either run the demo or our test programs)
+ * It has been slightly modified from the version given in the lab 2 documents to fit this lab
+ *
+ * {@value #leftMotor}  value from the left motor
+ * {@value #rightMotor} value from the right motor
+ * {@value #WHEEL_RAD}  constant, value of the wheel's radius
+ * {@value #TRACK}      constant, value of track
+ * {@value #SPEED}      constant, moving speed of robot
+ * {@value #usPort}     value from usPort
+ * 
+ * @author Maxime Bourassa
+ * @author Violet Wei
+ *
+ */
+public class Lab4 {
+
+  // Motor Objects, and Robot related constant parameters
+  //These all have getters to be accessed from other classes
+  public static final EV3LargeRegulatedMotor leftMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("D"));
+  public static final EV3LargeRegulatedMotor rightMotor =
+      new EV3LargeRegulatedMotor(LocalEV3.get().getPort("A"));
+  public static NavigationLab4 Nav = new NavigationLab4();
+  public static final TextLCD LCD = LocalEV3.get().getTextLCD();
+  public static Odometer odometer = new Odometer();
+  public static final double WHEEL_RAD = 2.1;
+  public static final double TRACK = 11.1;
+  public static final int SPEED = 75;
+
+  /**
+   * Localization type that can be used
+   * <li>{@link #FALLING_EDGE}</li>
+   * <li>{@link #RISING_EDGE}</li>     
+   */
+  public enum LocalizationType { FALLING_EDGE, RISING_EDGE };

```

Comment:
You're Javadoc `@link`ing to an element that has no Javadoc. Again, you can see this by hovering over `LocalizationType`. You should add brief Javadocs for the localization type as follows:

```java
/** ... */  
public enum LocalizationType {
  /** Falling edge ultrasonic localization. */ FALLING_EDGE, 
  /** Rising edge ultrasonic localization. */ RISING_EDGE
};
```